aws_lambda_builders/workflow.py

```diff
@@ -229,12 +257,31 @@ def __init__(
         self.architecture = architecture
         self.is_building_layer = is_building_layer
         self.experimental_flags = experimental_flags if experimental_flags else []
+
         self.build_in_source = build_in_source
+        if build_in_source is None:
+            self.build_in_source = self.BUILD_IN_SOURCE_BY_DEFAULT
+        self._validate_build_in_source()
 
         # Actions are registered by the subclasses as they seem fit
         self.actions = []
         self._binaries = {}
 
+    def _validate_build_in_source(self):
+        """
+        Validates that the value of build_in_source is supported by the chosen workflow.
+        """
+        if self.build_in_source not in self.BUILD_IN_SOURCE_SUPPORT.value:
+            error_reason = ""
+            if self.BUILD_IN_SOURCE_SUPPORT == BuildInSourceSupport.NOT_SUPPORTED:
+                error_reason = "This workflow does not support building in source"
+            elif self.BUILD_IN_SOURCE_SUPPORT == BuildInSourceSupport.EXCLUSIVELY_SUPPORTED:
+                error_reason = "This workflow only supports building in source"
+            else:
+                error_reason = "Unsupported value for build_in_source"
+
+            raise WorkflowFailedError(workflow_name=self.NAME, action_name=None, reason=error_reason)
```

Comment:
hmmm actually we might not want to fail. I might speak with UX about this but I think I'll change this to show a warning/log and use the default value for the workflow.

Reply:
For context: one of the reasons we might not want this to fail is that the `--in-source` flag is applied to all functions and layers of a project for now. If there's one function/layer that doesn't support building in source, this would fail the whole build. If it just raised a warning and used the default value, users would still be able to build (although there might be less visibility into the warning message vs. an error message)

Reply:
update: confirmed this with PM and changed logic to show warning LOG instead of erroring out.

This is what it would look like if custom make didn't support building in source:
<img width="632" alt="Screen Shot 2022-12-21 at 1 55 11 PM" src="https://user-images.githubusercontent.com/86501267/209010278-8ead2b63-8810-4da3-9455-42d32d645a29.png">